### PR TITLE
Feat: Added Memoize Func

### PR DESCRIPTION
**Pull request by MrDoomy**
Hello again 👋
I added the memo algorithm inside the **cache** folder (unit tests included)
If that's okay for you, can you add the `hacktoberfest-accepted` mention? 🙏

### CODE REVIEW

File: Cache/Memoize.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Memoize
+ * @param {Function} fn
+ * @returns
+ */
+export const memoize = (func) => {
+  // eslint-disable-next-line no-console
+  console.log(`Creating cache for function '${func.name}'`)
```
Comment (trasherdk) [inline Cache/Memoize.js:8]: Those `console.log()` are probably not crucial for the functionality, and don't look good when running tests.

Comment (trasherdk) [reply Cache/Memoize.js:8]: While on the test subject. Where did you get the idea to name the directory `__test__`, when the rest is just `test` ?

Comment (MrDoomy) [reply Cache/Memoize.js:8]: Oh! You're right. I apply my own convention rather than yours... My apologize 🙏 
I'm gonna fix it!

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Please add some comments explaining what the algorithm does and how.

**Review by raklaptudirm — APPROVED** [review]

**Review by pomkarnath98 — APPROVED** [review]

**Pull request merged**